src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -193,23 +193,47 @@ public StreamReader(string path, Encoding encoding, bool detectEncodingFromByteO
         {
         }
 
-        public StreamReader(string path, Encoding encoding, bool detectEncodingFromByteOrderMarks, int bufferSize) :
-            this(ValidateArgsAndOpenPath(path, encoding, bufferSize), encoding, detectEncodingFromByteOrderMarks, bufferSize, leaveOpen: false)
+        public StreamReader(string path, Encoding encoding, bool detectEncodingFromByteOrderMarks, int bufferSize)
+            : this(ValidateArgsAndOpenPath(path, encoding, bufferSize), encoding, detectEncodingFromByteOrderMarks, bufferSize, leaveOpen: false)
         {
         }
 
+        public StreamReader(string path, FileStreamOptions options)
+            : this(path, Encoding.UTF8, true, options)
+        {
+        }
+
+        public StreamReader(string path, Encoding encoding, bool detectEncodingFromByteOrderMarks, FileStreamOptions options)
+            : this(ValidateArgsAndOpenPath(path, encoding, options), encoding, detectEncodingFromByteOrderMarks, DefaultBufferSize)
+        {
+        }
+
+        private static Stream ValidateArgsAndOpenPath(string path, Encoding encoding, FileStreamOptions options)
+        {
+            ValidateArgs(path, encoding);
+            if (options == null)
+                throw new ArgumentNullException(nameof(options));
+
+            return new FileStream(path, options);
+        }
+
         private static Stream ValidateArgsAndOpenPath(string path, Encoding encoding, int bufferSize)
+        {
+            ValidateArgs(path, encoding);
+            if (bufferSize <= 0)
```

Comment:
Did we want to make this < rather than <= ?  Or that'll be handled separately?

Reply:
IMHO it should be handled separately in #53497